tools/online_editor/src/index.ts

```diff
@@ -307,6 +308,27 @@ function main() {
       },
       { mode: "split-bottom", ref: "Preview" },
     ],
+    [
+      () => {
+        const outline = new OutlineWidget(() => {
+          return [editor.language_client, editor.current_text_document_uri];
+        });
+
+        outline.on_goto_position = (
+          uri: string,
+          sl: number,
```

Comment:
is that a common pattern?  otherwise, i suggest using or defining a Position type that has a line and column.
Is something like https://docs.rs/lsp-types/latest/lsp_types/struct.Range.html  available in there?